File: analytical_engine/java/graphx-on-graphscope/src/main/java/com/alibaba/graphscope/graph/VertexDataManager.java
package com.alibaba.graphscope.graph;

import com.alibaba.graphscope.fragment.IFragment;
import com.alibaba.graphscope.graph.impl.VertexDataManagerImpl;
import java.util.List;
import org.apache.hadoop.io.Writable;

public interface VertexDataManager<VDATA_T> {
    void init(IFragment<Long,Long,VDATA_T,?> iFragment);

    VDATA_T getVertexData(long lid);

    void setVertexData(long lid, VDATA_T vertexData);

    <VDATA_T2> VertexDataManager<VDATA_T2> withNewVertexData(List<VDATA_T2> newVertexData);

    void setIFragment(IFragment<Long,Long,VDATA_T,?> fragment);

    void setValues(List<VDATA_T> vdatas);
}

File: analytical_engine/java/graphx-on-graphscope/src/main/scala/org/apache/spark/graphx/traits/VertexDataManager.scala
package org.apache.spark.graphx.traits

import com.alibaba.graphscope.utils.MappedBuffer

import scala.reflect.ClassTag


trait VertexDataManager[VD] {
  def getVDClz : Class[VD]
  def getVertexData(lid: Long): VD

  def setVertexData(lid: Long, vertexData: VD): Unit

  def withNewVertexData[VDATA_T2 : ClassTag](newVertexData: Array[VDATA_T2]): VertexDataManager[VDATA_T2]

  def setValues(vdatas: Array[VD]): Unit

  def writeBackVertexData(buffer : MappedBuffer) : Unit
}
